// @delhivery/react-dashboard-boilerplate@1.0.0/dist/index.js
var ye = Object.defineProperty;
var re = (t) => {
  throw TypeError(t);
};
var Se = (t, s, r) => s in t ? ye(t, s, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[s] = r;
var D = (t, s, r) => Se(t, typeof s != "symbol" ? s + "" : s, r), ae = (t, s, r) => s.has(t) || re("Cannot " + r);
var V = (t, s, r) => (ae(t, s, "read from private field"), r ? r.call(t) : s.get(t)), G = (t, s, r) => s.has(t) ? re("Cannot add the same private member more than once") : s instanceof WeakSet ? s.add(t) : s.set(t, r), P = (t, s, r, n) => (ae(t, s, "write to private field"), n ? n.call(t, r) : s.set(t, r), r);
import { jsx as e, jsxs as a, Fragment as Q } from "react/jsx-runtime";
import { useState as g, useCallback as E, createContext as ie, useContext as le, useEffect as F, useMemo as _e, useRef as ke } from "react";
import $ from "axios";
import { UserManager as X } from "oidc-client-ts";
import { useNavigate as de, useLocation as me, Route as ne } from "react-router-dom";
var T, q, H;
class Ie {
  constructor() {
    G(this, T, "");
    G(this, q, "");
    G(this, H, "");
  }
  setConfig(s) {
    P(this, T, s.clientId), P(this, q, s.landing_page), P(this, H, s.default_issuer);
  }
  set clientId(s) {
    P(this, T, s);
  }
  get clientId() {
    return V(this, T);
  }
  set landing_page(s) {
    P(this, q, s);
  }
  get landing_page() {
    return V(this, q) || "/";
  }
  set default_issuer(s) {
    P(this, H, s);
  }
  get default_issuer() {
    return V(this, H);
  }
}
T = new WeakMap(), q = new WeakMap(), H = new WeakMap();
const C = new Ie();
class Ue {
  configOptions(s, r, n) {
    const i = new URL(window.location.href), c = i.host, u = `${i.protocol}//${c}`;
    return {
      authority: s,
      client_id: r,
      redirect_uri: n,
      response_type: "code",
      scope: "openid profile email",
      automaticSilentRenew: !0,
      silent_redirect_uri: n,
      post_logout_redirect_uri: u,
      monitorSession: !0
      // oidc-client-ts automatically stores user data in sessionStorage
      // with key pattern: oidc.user:{authority}:{client_id}
    };
  }
}
const J = new Ue();
let Z = null;
const Ae = (t) => {
  Z = t;
};
let ee = null;
const O = (t) => {
  ee = t;
};
let oe = !1;
const ce = () => {
  oe || (oe = !0, $.interceptors.request.use(
    async (t) => {
      if (t.headers && (t.headers["X-Skip-Intercept"] || t.headers["x-skip-intercept"])) {
        const n = { ...t };
        return n.headers && (delete n.headers["X-Skip-Intercept"], delete n.headers["x-skip-intercept"]), n;
      }
      const s = t.headers || {};
      if (!(s.Authorization || s.authorization)) {
        let n = null;
        if (ee)
          try {
            const i = await ee.getUser();
            i && i.access_token && (n = i.access_token, n && sessionStorage.setItem("access_token", n));
          } catch (i) {
            console.warn("[Auth Interceptor] Failed to get user from UserManager:", i);
          }
        if (n || (n = sessionStorage.getItem("access_token")), n) {
          t.headers || (t.headers = {});
          const i = `Bearer ${n}`;
          t.headers.Authorization = i, t.headers.common || (t.headers.common = {}), t.headers.common.Authorization = i;
        }
      }
      return t;
    },
    (t) => Promise.reject(t)
  ), $.interceptors.response.use(
    (t) => t,
    async (t) => {
      var r, n, i;
      if (((r = t.response) == null ? void 0 : r.status) === 401) {
        const c = t.response.headers["www-authenticate"], m = c == null ? void 0 : c.match(/as_uri="([^"]+)"/), u = c == null ? void 0 : c.match(/realm="([^"]+)"/), h = u ? u[1] : " ", k = m ? m[1] + "/realms/" + h : C.default_issuer, N = new URL(window.location.href), I = N.host, U = `${N.protocol}//${I}/lib-auth-oauth-callback`;
        Z && Z(k, C.clientId, U).catch(() => {
        });
      } else if (((n = t.response) == null ? void 0 : n.status) === 400 && t.message === "Session not active")
        sessionStorage.clear();
      else
        return Promise.reject(t);
      const s = ((i = t.response) == null ? void 0 : i.status) === 401 ? t.message : "";
      return Promise.reject(new Error(s));
    }
  ));
};
class Ce {
  constructor() {
    D(this, "userManager", null);
    D(this, "loadDiscoveryDocument", async () => {
      const s = sessionStorage.getItem("issuer") ?? C.default_issuer, r = C.clientId, n = new URL(window.location.href), i = n.host, m = `${n.protocol}//${i}/lib-auth-oauth-callback`, u = J.configOptions(s, r, m);
      this.userManager = new X(u), O(this.userManager), await this.userManager.metadataService.getMetadata();
      const h = await this.userManager.getUser();
      h && h.access_token && sessionStorage.setItem("access_token", h.access_token);
    });
    D(this, "configureSSO", async (s, r, n) => {
      const i = J.configOptions(s, r, n);
      sessionStorage.setItem("issuer", s), this.userManager = new X(i), O(this.userManager), await this.userManager.metadataService.getMetadata();
      const c = await this.userManager.getUser();
      if (c && c.access_token) {
        const m = c.access_token;
        m && sessionStorage.setItem("access_token", m);
      } else
        console.error("No valid AT"), await this.userManager.signinRedirect();
    });
    D(this, "login", async (s, r, n) => {
      localStorage.setItem("loginInitiated", "true"), await this.configureSSO(s, r, n);
    });
    D(this, "clearStorage", () => {
      sessionStorage.clear(), localStorage.clear(), window.location.href = "/";
    });
    D(this, "logout", async () => {
      var s;
      try {
        const r = sessionStorage.getItem("issuer") || C.default_issuer, n = `oidc.user:${r}:`, i = sessionStorage.getItem(n);
        if (this.userManager && i && r) {
          const c = JSON.parse(i);
          await this.userManager.signoutRedirect({
            id_token_hint: c.id_token
          }), this.clearStorage();
          return;
        } else if (this.userManager) {
          await ((s = this.userManager) == null ? void 0 : s.signoutRedirect()), this.clearStorage();
          return;
        } else {
          console.error("No user manager found! Cannot logout."), this.clearStorage();
          return;
        }
      } catch (r) {
        console.error("Logout error:", r), this.clearStorage();
        return;
      }
    });
    D(this, "getUserManager", () => this.userManager);
    D(this, "setUserManager", (s) => {
      this.userManager = s, O(s);
    });
    D(this, "getAccessToken", async () => {
      if (this.userManager)
        try {
          const s = await this.userManager.getUser();
          if (s != null && s.access_token)
            return s.access_token;
        } catch {
        }
      return sessionStorage.getItem("access_token");
    });
    D(this, "hasValidAccessToken", () => sessionStorage.getItem("access_token") ? !0 : (this.userManager, !1));
  }
}
const A = new Ce();
Ae(
  (t, s, r) => A.login(t, s, r)
);
const Y = {
  title: "",
  logoSrc: "assets/Logo.svg",
  logoAlt: "Delhivery",
  navItems: []
}, ue = ie(void 0), Me = ({ children: t }) => {
  const [s, r] = g(Y), n = E((m) => {
    r((u) => ({ ...Y, ...u, ...m }));
  }, []), i = E(() => {
    r(Y);
  }, []), c = {
    headerConfig: s,
    setHeaderConfig: n,
    resetHeader: i
  };
  return /* @__PURE__ */ e(ue.Provider, { value: c, children: t });
}, De = () => {
  const t = le(ue);
  if (t === void 0)
    throw new Error("useHeaderConfig must be used within a HeaderConfigProvider");
  return t;
}, fe = ie(void 0), Be = ({ config: t, children: s }) => {
  const [r, n] = g(null), [i, c] = g(!0), [m, u] = g(null);
  F(() => {
    C.setConfig(t);
  }, [t]), F(() => {
    ce();
  }, []);
  const h = E(async (U, w, f) => {
    await A.login(U, w, f);
    const x = A.getUserManager();
    if (x) {
      u(x);
      const b = await x.getUser();
      b && (n(b), b.access_token && sessionStorage.setItem("access_token", b.access_token));
    }
  }, []);
  F(() => {
    (async () => {
      try {
        window.location.pathname.includes("/lib-auth-oauth-callback") || await A.loadDiscoveryDocument();
        let f = A.getUserManager();
        if (!f) {
          const x = sessionStorage.getItem("issuer") ?? t.default_issuer, b = t.clientId, M = new URL(window.location.href), j = M.host, y = `${M.protocol}//${j}/lib-auth-oauth-callback`, K = J.configOptions(x, b, y);
          f = new X(K), A.setUserManager(f);
        }
        if (f) {
          u(f), O(f);
          const x = await f.getUser();
          if (x && x.access_token)
            n(x), sessionStorage.setItem("access_token", x.access_token);
          else if (sessionStorage.getItem("access_token") && !x)
            try {
              const M = await f.getUser();
              M && M.access_token && (n(M), sessionStorage.setItem("access_token", M.access_token));
            } catch {
              console.warn("[AuthContext] Failed to get user, but token exists in sessionStorage");
            }
          f.events.addUserLoaded((b) => {
            n(b), b.access_token && sessionStorage.setItem("access_token", b.access_token);
          }), f.events.addUserSignedOut(() => {
            n(null), O(null);
          });
        }
        ce();
      } catch (w) {
        console.error("Auth initialization error:", w);
      } finally {
        c(!1);
      }
    })();
  }, [t]);
  const k = E(async () => {
    await A.logout(), n(null), u(null), O(null);
  }, []), N = E(() => m || A.getUserManager(), [m]), I = {
    user: r,
    isAuthenticated: !!r && !!r.access_token,
    isLoading: i,
    login: h,
    logout: k,
    getUserManager: N
  }, p = _e(() => I, [r, i, h, k, N]);
  return /* @__PURE__ */ e(fe.Provider, { value: p, children: /* @__PURE__ */ a(Me, { children: [
    /* @__PURE__ */ e(Pe, {}),
    s
  ] }) });
}, W = () => {
  const t = le(fe);
  if (t === void 0)
    throw new Error("useAuth must be used within an AuthProvider");
  return t;
}, je = ({ firstName: t, lastName: s, email: r }) => {
  const { logout: n } = W(), i = de(), [c, m] = g(!1), u = (w) => w ? w.charAt(0).toUpperCase() + w.slice(1).toLowerCase() : "", h = () => !t && !s ? "User" : s ? u(t) + " " + u(s) : u(t), k = () => {
    if (!t && !s)
      return "U";
    const w = t.charAt(0);
    if (!s)
      return u(w);
    const f = s.charAt(0);
    return u(w) + u(f);
  }, N = () => r || "user@example.com", I = async () => {
    await n();
  }, p = () => {
    m(!1), i("/lib-user-profile");
  };
  return /* @__PURE__ */ a("div", { children: [
    /* @__PURE__ */ a(
      "button",
      {
        id: "dropdownMenu",
        type: "button",
        className: `flex gap-2.5 items-center p-1 ${c ? "shadow-[0_20px_56px_-12px_rgba(0,0,0,0.25)]" : ""}`,
        onClick: () => {
          m(!c);
        },
        children: [
          /* @__PURE__ */ e("div", { className: "flex p-2", children: /* @__PURE__ */ a("div", { className: "flex items-center gap-2", children: [
            /* @__PURE__ */ e("div", { className: "flex py-3 pr-2 pl-[9px] rounded-[20px] bg-frame-bg-300", children: /* @__PURE__ */ e("span", { className: "font-medium text-base/[16px] text-color-font-600 font-sans", children: k() }) }),
            /* @__PURE__ */ a("div", { className: "flex flex-col items-start my-0.5", children: [
              /* @__PURE__ */ e("div", { className: "flex basis-full justify-start items-center", children: /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600 font-sans", children: h() }) }),
              /* @__PURE__ */ e("div", { className: "flex basis-full justify-start items-center", children: /* @__PURE__ */ e("span", { className: "text-xs font-medium text-color-font-400 text-ellipsis overflow-hidden font-sans", children: N() }) })
            ] })
          ] }) }),
          /* @__PURE__ */ e("div", { className: "p-1", children: /* @__PURE__ */ e("img", { src: "assets/down-arrow.svg", alt: "dropdown" }) })
        ]
      }
    ),
    c && /* @__PURE__ */ a("div", { className: "flex flex-col absolute w-[306.93px] rounded-b-md shadow-[0_20px_56px_-12px_rgba(0,0,0,0.25)] bg-frame-bg-100 overflow-visible z-50", children: [
      /* @__PURE__ */ e("div", { className: "flex px-4 py-2 items-center self-stretch border-b border-b-color-font-300", children: /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: "text-color-font-600 font-medium block text-sm font-sans",
          onClick: p,
          children: "Edit Profile"
        }
      ) }),
      /* @__PURE__ */ e("div", { className: "flex px-4 py-2 items-center self-stretch", children: /* @__PURE__ */ e(
        "button",
        {
          type: "submit",
          className: "text-color-font-600 font-medium block text-sm font-sans",
          onClick: I,
          children: "Logout"
        }
      ) })
    ] })
  ] });
}, $e = ({ items: t }) => me().pathname === "/lib-user-profile" ? /* @__PURE__ */ e("div", { className: "flex items-center pl-8 pr-6", children: /* @__PURE__ */ a("div", { className: "flex items-center justify-center gap-4", children: [
  /* @__PURE__ */ e(
    "a",
    {
      href: "/lib-user-profile",
      className: "text-sm font-medium hover:text-blue-600 text-color-font-600",
      children: "Personal Info"
    }
  ),
  /* @__PURE__ */ e("div", { className: "self-stretch", children: /* @__PURE__ */ e("img", { src: "assets/breadcrumb.svg", alt: "breadcrumb-arrow" }) })
] }) }) : t && t.length > 0 ? /* @__PURE__ */ e("div", { className: "flex items-center px-6", children: /* @__PURE__ */ e("div", { className: "flex items-center justify-center gap-4", children: t.map((r, n) => /* @__PURE__ */ a("div", { className: "flex items-center gap-4", children: [
  /* @__PURE__ */ e(
    "a",
    {
      href: r.path,
      className: "text-sm font-medium hover:text-blue-600 text-color-font-600",
      children: r.label
    }
  ),
  n < t.length - 1 && /* @__PURE__ */ e("div", { className: "self-stretch", children: /* @__PURE__ */ e("img", { src: "assets/breadcrumb.svg", alt: "breadcrumb-arrow" }) })
] }, r.path)) }) }) : null, Le = ({ navItems: t }) => {
  const s = me(), r = (n) => s.pathname === n;
  return !t || t.length === 0 ? null : /* @__PURE__ */ e("div", { className: "flex items-center justify-center gap-4", children: t.map((n) => /* @__PURE__ */ e(
    "a",
    {
      href: n.route,
      className: `text-sm font-bold px-4 py-2 text-color-font-600 font-sans transition-all duration-200 border rounded-md ${r(n.route) ? "text-red-600 border-red-600 bg-red-50" : "text-color-font-600 border-transparent hover:border-color-font-300 hover:bg-gray-200"}`,
      children: n.label
    },
    n.route
  )) });
}, Pe = () => {
  var I;
  const { isAuthenticated: t, isLoading: s } = W(), { headerConfig: r } = De(), [n, i] = g(""), [c, m] = g(""), [u, h] = g(""), k = () => A.hasValidAccessToken() || !!sessionStorage.getItem("access_token"), N = E(async () => {
    var U;
    const p = sessionStorage.getItem("issuer");
    if (p) {
      if (!sessionStorage.getItem("access_token")) {
        console.warn("[Header] No access token available, skipping account fetch");
        return;
      }
      try {
        const f = p.startsWith("http") ? p.replace(/^https?:\/\/[^/]+/, "") + "/account" : `${p}/account`, x = await $.get(f);
        i(x.data.firstName), m(x.data.lastName || ""), h(x.data.email);
      } catch (f) {
        console.error("Error fetching user data:", f), ((U = f.response) == null ? void 0 : U.status) === 401 && (console.error("[Header] 401 Unauthorized"), console.error("[Header] Token exists:", !!sessionStorage.getItem("access_token")), console.error("[Header] Request URL:", `${p}/account`));
      }
    }
  }, []);
  return F(() => {
    if (!s)
      if (k())
        N();
      else {
        const p = A.getUserManager();
        if (p) {
          const U = () => {
            N();
          };
          return p.events.addUserLoaded(U), p.events.addAccessTokenExpiring(() => {
          }), () => {
            p.events.removeUserLoaded(U);
          };
        }
      }
  }, [t, s, N]), k() ? /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e("nav", { className: "bg-white", children: /* @__PURE__ */ a("div", { className: "flex items-center justify-between self-stretch w-full border-b border-color-font-300", children: [
    /* @__PURE__ */ a("div", { className: "h-16 flex items-center gap-6", children: [
      /* @__PURE__ */ e("div", { className: "flex items-start border-r border-color-font-300 pt-4 pb-3.5 max-w-64 h-auto", children: /* @__PURE__ */ e("div", { className: "flex justify-center items-center gap-2 self-stretch px-6 w-full", children: /* @__PURE__ */ e("div", { className: "flex pt-[5px] pb-[4.12px] justify-center items-center", children: /* @__PURE__ */ e("a", { href: "/", children: /* @__PURE__ */ e(
        "img",
        {
          className: "h-6 w-36",
          src: (r == null ? void 0 : r.logoSrc) || "assets/Logo.svg",
          alt: (r == null ? void 0 : r.logoAlt) || "Delhivery"
        }
      ) }) }) }) }),
      /* @__PURE__ */ e("div", { className: "flex items-center", children: /* @__PURE__ */ e("span", { className: "text-2xl font-black text-gray-900 flex font-sans leading-relaxed", children: r == null ? void 0 : r.title }) }),
      /* @__PURE__ */ e($e, {})
    ] }),
    /* @__PURE__ */ e("div", { className: "flex items-center justify-between pl-[13px]", children: /* @__PURE__ */ a("div", { className: "flex items-center justify-end gap-2.5", children: [
      r != null && r.navItems && ((I = r == null ? void 0 : r.navItems) == null ? void 0 : I.length) > 0 ? /* @__PURE__ */ e(Le, { navItems: r == null ? void 0 : r.navItems }) : null,
      r != null && r.otherComponent ? /* @__PURE__ */ e("div", { className: "flex items-center", children: r.otherComponent }) : null,
      /* @__PURE__ */ e(je, { firstName: n, lastName: c, email: u })
    ] }) })
  ] }) }) }) : null;
}, Oe = () => {
  const { getUserManager: t, isLoading: s } = W(), r = () => A.hasValidAccessToken() || !!sessionStorage.getItem("access_token"), [n, i] = g({
    username: "",
    email: "",
    firstName: "",
    lastName: ""
  }), [c, m] = g("Signing In"), [u, h] = g(!1), [k, N] = g(""), [I, p] = g(!1), [U, w] = g(""), [f, x] = g([]), [b, M] = g([]), [j, te] = g([]), [y, K] = g([]);
  F(() => {
    s || r() && (ge(), he(), pe());
  }, [s]);
  const ge = async () => {
    const o = sessionStorage.getItem("issuer");
    if (o)
      try {
        const l = o.startsWith("http") ? o.replace(/^https?:\/\/[^/]+/, "") + "/account" : `${o}/account`, d = await $.get(l);
        if (i({
          username: d.data.username,
          email: d.data.email,
          firstName: d.data.firstName,
          lastName: d.data.lastName || ""
        }), d.data.userProfileMetadata) {
          const S = [], L = [];
          d.data.userProfileMetadata.attributes.forEach((v) => {
            v.required && S.push(v.name), v.readOnly && L.push(v.name);
          }), te(S), K(L);
        }
      } catch (l) {
        console.error("Error fetching user data:", l);
      }
  }, se = async (o) => {
    o.preventDefault();
    const l = sessionStorage.getItem("issuer");
    if (l)
      try {
        const d = l.startsWith("http") ? l.replace(/^https?:\/\/[^/]+/, "") + "/account/" : `${l}/account/`;
        await $.post(d, n), window.location.reload();
      } catch (d) {
        console.error("Error saving user data:", d);
      }
  }, B = (o, l) => {
    i((d) => ({ ...d, [o]: l }));
  }, he = async () => {
    var l, d, S, L;
    const o = sessionStorage.getItem("issuer");
    if (o)
      try {
        const v = o.startsWith("http") ? o.replace(/^https?:\/\/[^/]+/, "") + "/account/credentials" : `${o}/account/credentials`, R = (await $.get(v)).data;
        if ((d = (l = R[0]) == null ? void 0 : l.userCredentialMetadatas) != null && d[0]) {
          const z = R[0].userCredentialMetadatas[0].credential.createdDate;
          N(new Date(z).toString().slice(0, -22)), h(!0);
        }
        if ((L = (S = R[1]) == null ? void 0 : S.userCredentialMetadatas) != null && L[0]) {
          const z = R[1].userCredentialMetadatas[0].credential.createdDate;
          w(new Date(z).toString().slice(0, -22)), p(!0);
        }
      } catch (v) {
        console.error("Error fetching password/OTP details:", v);
      }
  }, pe = async () => {
    const o = sessionStorage.getItem("issuer");
    if (o)
      try {
        const l = o.startsWith("http") ? o.replace(/^https?:\/\/[^/]+/, "") + "/account/sessions/devices" : `${o}/account/sessions/devices`, d = await $.get(l), S = [];
        let L = !1;
        if (d.data.forEach((v) => {
          v.sessions.forEach((_) => {
            const R = {
              device: `${v.os} / ${_.browser}`,
              current_session: _.current,
              ip: _.ipAddress,
              started: new Date(_.started * 1e3).toString().slice(0, -22),
              last_accessed: new Date(_.lastAccess * 1e3).toString().slice(0, -22),
              expires: new Date(_.expires * 1e3).toString().slice(0, -22),
              clients: _.clients.map((z) => z.clientName),
              id: _.id
            };
            _.current && (L = !0), S.push(R);
          });
        }), S.sort((v, _) => _.current_session ? 1 : -1), x(S), M(new Array(S.length).fill(!0).map((v, _) => _ !== 0)), !L) {
          const v = t();
          v && (await v.signoutRedirect(), sessionStorage.clear(), localStorage.clear(), window.location.href = "/");
        }
      } catch (l) {
        console.error("Error fetching device details:", l);
      }
  }, xe = () => {
    const o = sessionStorage.getItem("issuer"), l = C.clientId, d = new URL(location.href);
    return `${o}/protocol/openid-connect/auth?client_id=${l}&redirect_uri=${d.protocol}//${d.host}/lib-auth-oauth-callback&response_type=code&scope=openid&kc_action=UPDATE_PASSWORD`;
  }, be = () => {
    const o = sessionStorage.getItem("issuer"), l = C.clientId, d = new URL(location.href);
    return `${o}/protocol/openid-connect/auth?client_id=${l}&redirect_uri=${d.protocol}//${d.host}/lib-auth-oauth-callback&response_type=code&scope=openid&kc_action=CONFIGURE_TOTP`;
  }, ve = (o) => {
    const l = [...b];
    for (let d = 0; d < f.length; d++)
      d !== o && (l[d] = !0);
    l[o] = !l[o], M(l);
  }, Ne = () => {
    window.location.reload();
  }, we = async (o) => {
    const l = f[o];
    if (l.current_session === !0) {
      const d = t();
      d && (await d.signoutRedirect(), sessionStorage.clear(), localStorage.clear(), window.location.href = "/");
    } else {
      const d = sessionStorage.getItem("issuer");
      if (d)
        try {
          const S = d.startsWith("http") ? d.replace(/^https?:\/\/[^/]+/, "") + `/account/sessions/${l.id}` : `${d}/account/sessions/${l.id}`;
          await $.delete(S), window.location.reload();
        } catch (S) {
          console.error("Error signing out of session:", S);
        }
    }
  };
  return r() ? /* @__PURE__ */ a("div", { className: "flex flex-col items-center gap-6", children: [
    /* @__PURE__ */ a("div", { className: "flex flex-col w-2/3 gap-6 mt-8", children: [
      /* @__PURE__ */ a("div", { className: "flex flex-col gap-6 items-start", children: [
        /* @__PURE__ */ a("div", { className: "flex items-start gap-2", children: [
          /* @__PURE__ */ e("div", { className: "flex justify-center items-center p-2 border rounded-lg border-frame-bg-300", children: /* @__PURE__ */ e("img", { className: "self-stretch", src: "assets/user-icon.svg", alt: "user" }) }),
          /* @__PURE__ */ a("div", { className: "flex flex-col justify-center items-start text-color-font-600 gap-1", children: [
            /* @__PURE__ */ e("span", { className: "text-lg/[18px] font-medium", children: "Personal Info" }),
            /* @__PURE__ */ e("span", { className: "text-sm font-normal", children: "Manage your basic Information" })
          ] })
        ] }),
        /* @__PURE__ */ e("div", { className: "grid gap-4 w-full items-center", children: /* @__PURE__ */ a("form", { onSubmit: se, children: [
          /* @__PURE__ */ a("div", { className: "flex gap-4 justify-center items-start self-stretch", children: [
            /* @__PURE__ */ a("div", { className: "flex flex-col basis-1/2 gap-1 items-start self-stretch", children: [
              /* @__PURE__ */ a("label", { className: "flex self-stretch items-center text-color-font-600 text-sm font-medium", htmlFor: "username", children: [
                "Username",
                j.includes("username") && /* @__PURE__ */ e("span", { className: "required text-new-red-100 font-medium text-sm", children: " *" })
              ] }),
              /* @__PURE__ */ e(
                "div",
                {
                  className: `flex py-[9px] px-[13px] gap-1 border border-color-font-300 rounded-md w-full ${y.includes("username") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100 shadow-sm"}`,
                  children: /* @__PURE__ */ e(
                    "input",
                    {
                      className: `placeholder:text-color-font-400 text-sm font-normal w-full focus:outline-none text-color-font-600 ${y.includes("username") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100"}`,
                      type: "text",
                      placeholder: "Enter Username",
                      value: n.username,
                      onChange: (o) => B("username", o.target.value),
                      readOnly: y.includes("username"),
                      required: j.includes("username")
                    }
                  )
                }
              )
            ] }),
            /* @__PURE__ */ a("div", { className: "flex flex-col basis-1/2 gap-1 items-start self-stretch", children: [
              /* @__PURE__ */ a("label", { className: "flex self-stretch items-center text-color-font-600 text-sm font-medium", htmlFor: "email", children: [
                "Email ID",
                j.includes("email") && /* @__PURE__ */ e("span", { className: "required text-new-red-100 font-medium text-sm", children: " *" })
              ] }),
              /* @__PURE__ */ e(
                "div",
                {
                  className: `flex py-[9px] px-[13px] gap-1 border border-color-font-300 rounded-md w-full ${y.includes("email") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100 shadow-sm"}`,
                  children: /* @__PURE__ */ e(
                    "input",
                    {
                      className: `placeholder:text-color-font-400 text-sm font-normal w-full focus:outline-none text-color-font-600 ${y.includes("email") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100"}`,
                      type: "email",
                      placeholder: "Enter Email ID",
                      value: n.email,
                      onChange: (o) => B("email", o.target.value),
                      readOnly: y.includes("email"),
                      required: j.includes("email")
                    }
                  )
                }
              )
            ] })
          ] }),
          /* @__PURE__ */ a("div", { className: "flex gap-4 justify-center items-start self-stretch mt-4", children: [
            /* @__PURE__ */ a("div", { className: "flex flex-col basis-1/2 gap-1 items-start self-stretch", children: [
              /* @__PURE__ */ a("label", { className: "flex self-stretch items-center text-color-font-600 text-sm font-medium", htmlFor: "firstName", children: [
                "First Name",
                j.includes("firstName") && /* @__PURE__ */ e("span", { className: "required text-new-red-100 font-medium text-sm", children: " *" })
              ] }),
              /* @__PURE__ */ e(
                "div",
                {
                  className: `flex py-[9px] px-[13px] gap-1 border border-color-font-300 rounded-md w-full ${y.includes("firstName") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100 shadow-sm"}`,
                  children: /* @__PURE__ */ e(
                    "input",
                    {
                      className: `placeholder:text-color-font-400 text-sm font-normal w-full focus:outline-none text-color-font-600 ${y.includes("firstName") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100"}`,
                      type: "text",
                      placeholder: "Enter Firstname",
                      value: n.firstName,
                      onChange: (o) => B("firstName", o.target.value),
                      readOnly: y.includes("firstName"),
                      required: j.includes("firstName")
                    }
                  )
                }
              )
            ] }),
            /* @__PURE__ */ a("div", { className: "flex flex-col basis-1/2 gap-1 items-start self-stretch", children: [
              /* @__PURE__ */ a("label", { className: "flex self-stretch items-center text-color-font-600 text-sm font-medium", htmlFor: "lastName", children: [
                "Last Name",
                j.includes("lastName") && /* @__PURE__ */ e("span", { className: "required text-new-red-100 font-medium text-sm", children: " *" })
              ] }),
              /* @__PURE__ */ e(
                "div",
                {
                  className: `flex py-[9px] px-[13px] gap-1 border border-color-font-300 focus:border-primary-brand-200 rounded-md w-full ${y.includes("lastName") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100 shadow-sm"}`,
                  children: /* @__PURE__ */ e(
                    "input",
                    {
                      className: `placeholder:text-color-font-400 text-sm font-normal w-full focus:outline-none text-color-font-600 ${y.includes("lastName") ? "bg-frame-bg-300 cursor-not-allowed" : "bg-frame-bg-100"}`,
                      type: "text",
                      placeholder: "Enter Lastname",
                      value: n.lastName,
                      onChange: (o) => B("lastName", o.target.value),
                      readOnly: y.includes("lastName")
                    }
                  )
                }
              )
            ] })
          ] })
        ] }) })
      ] }),
      /* @__PURE__ */ a("div", { className: "grid py-4 items-start self-stretch gap-6", children: [
        /* @__PURE__ */ a("div", { className: "flex items-center gap-2 self-stretch", children: [
          /* @__PURE__ */ e("div", { className: "flex justify-center items-center p-2 border rounded-lg border-frame-bg-300", children: /* @__PURE__ */ e("img", { src: "assets/account-summary.svg", alt: "account" }) }),
          /* @__PURE__ */ e("div", { className: "flex", children: /* @__PURE__ */ e("span", { className: "font-medium text-color-font-600 text-lg/[18px]", children: "Account Summary" }) })
        ] }),
        /* @__PURE__ */ a("div", { className: "grid items-start self-stretch gap-4", children: [
          /* @__PURE__ */ a("div", { className: "flex self-stretch border-b border-frame-bg-300 gap-8", children: [
            /* @__PURE__ */ e(
              "button",
              {
                className: `flex flex-col text-sm px-1 pb-4 font-medium ${c === "Signing In" ? "text-delhivery-color-100 border-delhivery-color-100 border-b-2" : "text-color-font-600"}`,
                onClick: () => m("Signing In"),
                children: "Signing In"
              }
            ),
            /* @__PURE__ */ e(
              "button",
              {
                className: `flex flex-col text-sm px-1 pb-4 font-medium ${c === "Device Activity" ? "text-delhivery-color-100 border-delhivery-color-100 border-b-2" : "text-color-font-600"}`,
                onClick: () => m("Device Activity"),
                children: "Device Activity"
              }
            )
          ] }),
          /* @__PURE__ */ a("div", { id: "default-tab-content", children: [
            c === "Signing In" && /* @__PURE__ */ a("div", { className: "flex p-4 gap-3 items-center self-stretch rounded-md bg-color-font-200", children: [
              /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e("img", { src: "assets/info.svg", alt: "info" }) }),
              /* @__PURE__ */ e("div", { className: "flex items-start gap-3", children: /* @__PURE__ */ e("span", { className: "text-color-font-600 text-sm", children: "Configure ways to sign in" }) })
            ] }),
            c === "Device Activity" && /* @__PURE__ */ a("div", { className: "flex p-4 gap-3 items-center self-stretch rounded-md bg-color-font-200", children: [
              /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e("img", { src: "assets/info.svg", alt: "info" }) }),
              /* @__PURE__ */ e("div", { className: "flex items-start gap-3", children: /* @__PURE__ */ e("span", { className: "text-color-font-600 text-sm", children: "Sign out of any unfamiliar devices" }) })
            ] })
          ] })
        ] }),
        c === "Signing In" && /* @__PURE__ */ a("div", { className: "flex flex-col gap-4", children: [
          /* @__PURE__ */ a("div", { className: "flex flex-col gap-6 p-4 border rounded-lg self-stretch items-start border-bord-white-100", children: [
            /* @__PURE__ */ a("div", { className: "flex items-center gap-2", children: [
              /* @__PURE__ */ e("div", { className: "p-2 rounded-lg bg-frame-bg-200", children: /* @__PURE__ */ e("img", { src: "assets/basic_authentication.svg", alt: "auth" }) }),
              /* @__PURE__ */ e("span", { className: "flex text-color-font-600 text-lg font-medium", children: "Basic Authentication" }),
              !u && /* @__PURE__ */ a("div", { className: "flex px-2 py-0.5 justify-center items-center gap-1 rounded-xl bg-pending-bg-color-100", children: [
                /* @__PURE__ */ e("div", { className: "flex w-3 justify-center items-center", children: /* @__PURE__ */ e("img", { src: "assets/info-pending.svg", alt: "pending" }) }),
                /* @__PURE__ */ e("span", { className: "text-sm text-pending-color-100 font-medium", children: "Password is not setup" })
              ] })
            ] }),
            /* @__PURE__ */ a("div", { className: "flex items-center self-stretch gap-3", children: [
              /* @__PURE__ */ a("div", { className: "flex flex-col justify-center items-start gap-1 basis-3/4", children: [
                /* @__PURE__ */ e("span", { className: "flex justify-start items-center font-medium text-base text-color-font-600", children: "Password" }),
                u && /* @__PURE__ */ a("div", { className: "flex self-stretch items-start", children: [
                  /* @__PURE__ */ e("span", { className: "font-medium text-xs text-color-font-600", children: "Created On:  " }),
                  /* @__PURE__ */ e("span", { className: "font-normal text-xs text-color-font-600", children: k })
                ] })
              ] }),
              /* @__PURE__ */ e("div", { className: "flex justify-end items-center basis-1/4", children: /* @__PURE__ */ a(
                "a",
                {
                  role: "button",
                  href: xe(),
                  target: "_blank",
                  className: "flex pl-[15px] py-[9px] gap-2 pr-[17px] justify-center items-center rounded-md border border-bord-white-200 shadow-sm",
                  children: [
                    /* @__PURE__ */ e("img", { src: "assets/update-password.svg", alt: "update" }),
                    /* @__PURE__ */ e("span", { className: "text-color-font-700 text-sm font-medium", children: "Update Password" })
                  ]
                }
              ) })
            ] })
          ] }),
          /* @__PURE__ */ a("div", { className: "grid gap-6 p-4 items-start border border-bord-white-100 rounded-lg", children: [
            /* @__PURE__ */ a("div", { className: "flex gap-2 items-center", children: [
              /* @__PURE__ */ e("div", { className: "flex p-2 items-start rounded-lg bg-frame-bg-200", children: /* @__PURE__ */ e("img", { src: "assets/tfa.svg", alt: "tfa" }) }),
              /* @__PURE__ */ e("span", { className: "flex text-color-font-600 text-lg font-medium", children: "Two-Factor Authentication" }),
              !I && /* @__PURE__ */ a("div", { className: "flex px-2 py-0.5 justify-center items-center gap-1 rounded-xl bg-pending-bg-color-100", children: [
                /* @__PURE__ */ e("div", { className: "flex w-3 justify-center items-center", children: /* @__PURE__ */ e("img", { src: "assets/info-pending.svg", alt: "pending" }) }),
                /* @__PURE__ */ e("span", { className: "text-sm text-pending-color-100 font-medium", children: "Authenticator application is not setup" })
              ] })
            ] }),
            /* @__PURE__ */ a("div", { className: "flex gap-1 w-full self-stretch items-center", children: [
              /* @__PURE__ */ a("div", { className: "flex flex-col basis-2/3 gap-1 items-start", children: [
                /* @__PURE__ */ e("span", { className: "text-color-font-600 font-medium text-base", children: "Authenticator application" }),
                I && /* @__PURE__ */ a("div", { className: "flex self-stretch items-start", children: [
                  /* @__PURE__ */ e("span", { className: "font-medium text-xs text-color-font-600", children: "Created On:  " }),
                  /* @__PURE__ */ e("span", { className: "font-normal text-xs text-color-font-600", children: U })
                ] })
              ] }),
              /* @__PURE__ */ e("div", { className: "flex justify-end items-center basis-1/3", children: /* @__PURE__ */ a(
                "a",
                {
                  role: "button",
                  href: be(),
                  target: "_blank",
                  className: "flex pl-[15px] py-[9px] gap-2 pr-[17px] justify-center items-center rounded-md border border-bord-white-200 shadow-sm",
                  children: [
                    /* @__PURE__ */ e("img", { src: "assets/setup2factorauth.svg", alt: "setup" }),
                    /* @__PURE__ */ e("span", { className: "text-color-font-700 text-sm font-medium", children: "Set up authenticator application" })
                  ]
                }
              ) })
            ] })
          ] })
        ] }),
        c === "Device Activity" && /* @__PURE__ */ a("div", { className: "flex flex-col gap-4 items-start self-stretch", children: [
          /* @__PURE__ */ a("div", { className: "flex justify-between items-center self-stretch", children: [
            /* @__PURE__ */ e("span", { className: "text-base font-medium text-color-font-600", children: "Signed in devices" }),
            /* @__PURE__ */ a(
              "button",
              {
                className: "flex pl-[15px] pr-[17px] py-[9px] justify-center items-center gap-2 rounded-md border border-color-font-300 bg-frame-bg-100",
                onClick: Ne,
                children: [
                  /* @__PURE__ */ e("div", { className: "flex w-5 p-0.5 justify-center items-center", children: /* @__PURE__ */ e("img", { src: "assets/refresh-page.svg", alt: "refresh" }) }),
                  /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-700", children: "Refresh the page" })
                ]
              }
            )
          ] }),
          /* @__PURE__ */ e("div", { className: "border self-stretch border-frame-bg-300" }),
          f.map((o, l) => /* @__PURE__ */ a("div", { className: "flex flex-col gap-4", children: [
            /* @__PURE__ */ a("div", { className: "flex items-start gap-3 self-stretch", children: [
              /* @__PURE__ */ e("div", { className: "flex p-2 items-start rounded-lg", children: /* @__PURE__ */ e("div", { className: "flex items-center justify-center", children: /* @__PURE__ */ e("img", { src: "assets/signed-in-devices.svg", alt: "device" }) }) }),
              /* @__PURE__ */ a("div", { className: "flex flex-col items-start self-stretch w-full", children: [
                /* @__PURE__ */ a("div", { className: "flex py-2 px-3 items-center self-stretch gap-4 justify-between", children: [
                  /* @__PURE__ */ a("div", { className: "flex gap-4 items-center justify-center", children: [
                    /* @__PURE__ */ e("span", { className: "self-stretch text-sm font-medium text-color-font-600", children: o.device }),
                    /* @__PURE__ */ e("div", { className: "flex flex-col items-start gap-8 bg-frame-bg-100", children: o.current_session && /* @__PURE__ */ e("div", { className: "flex py-0.5 px-3 rounded-xl justify-center items-center bg-success-100", children: /* @__PURE__ */ e("span", { className: "text-sm font-medium text-success-200", children: "Current Session" }) }) })
                  ] }),
                  /* @__PURE__ */ a("div", { className: "flex items-start gap-4", children: [
                    /* @__PURE__ */ a(
                      "button",
                      {
                        className: "flex py-[9px] pl-[15px] pr-[17px] justify-center items-center gap-2 border border-frame-bg-300 bg-frame-bg-100 rounded-md",
                        onClick: () => we(l),
                        children: [
                          /* @__PURE__ */ e("div", { className: "flex w-5 p-0.5 justify-center items-center", children: /* @__PURE__ */ e("img", { className: "flex self-stretch", src: "assets/sign-out.svg", alt: "signout" }) }),
                          /* @__PURE__ */ e("span", { className: "flex text-sm font-medium text-color-font-700", children: "Sign out" })
                        ]
                      }
                    ),
                    /* @__PURE__ */ a(
                      "button",
                      {
                        className: "flex pl-[17px] pr-[15px] py-[9px] justify-center items-center gap-2 border border-frame-bg-300 bg-frame-bg-100 rounded-md",
                        onClick: () => ve(l),
                        children: [
                          /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-700", children: b[l] ? "More" : "Less" }),
                          /* @__PURE__ */ e("div", { className: "flex w-5 p-0.5 justify-center items-center", children: /* @__PURE__ */ e(
                            "img",
                            {
                              className: "flex self-stretch",
                              src: b[l] ? "assets/chevron-down.svg" : "assets/chevron-up.svg",
                              alt: "toggle"
                            }
                          ) })
                        ]
                      }
                    )
                  ] })
                ] }),
                !b[l] && /* @__PURE__ */ a("div", { className: "flex items-start self-stretch", children: [
                  /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                    /* @__PURE__ */ e("div", { className: "flex px-3 py-4 items-center self-stretch gap-2", children: /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                      /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600", children: "IP Address" }),
                      /* @__PURE__ */ e("span", { className: "text-sm font-normal text-color-font-500", children: o.ip })
                    ] }) }),
                    /* @__PURE__ */ e("div", { className: "flex px-3 py-4 items-center self-stretch gap-2", children: /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                      /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600", children: "Started" }),
                      /* @__PURE__ */ e("span", { className: "text-sm font-normal text-color-font-500", children: o.started })
                    ] }) })
                  ] }),
                  /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                    /* @__PURE__ */ e("div", { className: "flex px-3 py-4 items-center self-stretch gap-2", children: /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                      /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600", children: "Last Accessed" }),
                      /* @__PURE__ */ e("span", { className: "text-sm font-normal text-color-font-500", children: o.last_accessed })
                    ] }) }),
                    /* @__PURE__ */ e("div", { className: "flex px-3 py-4 items-center self-stretch gap-2", children: /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                      /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600", children: "Expires" }),
                      /* @__PURE__ */ e("span", { className: "text-sm font-normal text-color-font-500", children: o.expires })
                    ] }) })
                  ] }),
                  /* @__PURE__ */ e("div", { className: "flex flex-col items-start", children: /* @__PURE__ */ e("div", { className: "flex px-3 py-4 items-center self-stretch gap-2", children: /* @__PURE__ */ a("div", { className: "flex flex-col items-start", children: [
                    /* @__PURE__ */ e("span", { className: "text-sm font-medium text-color-font-600", children: "Clients" }),
                    /* @__PURE__ */ e("span", { className: "text-sm font-normal text-color-font-500", children: o.clients.join(", ") })
                  ] }) }) })
                ] })
              ] })
            ] }),
            /* @__PURE__ */ e("div", { className: "border self-stretch border-frame-bg-300" })
          ] }, o.id))
        ] })
      ] })
    ] }),
    /* @__PURE__ */ e("div", { className: "flex py-6 px-4 items-center justify-between border-t border-color-font-300 sticky bottom-0 bg-frame-bg-100 shadow-[0_-1px_0px_0px_rgba(0,0,0,0.08)] w-savebar", children: /* @__PURE__ */ a("div", { className: "flex basis-full items-center justify-end gap-3", children: [
      /* @__PURE__ */ e("button", { className: "flex px-[17px] py-[9px] justify-center items-center rounded-md border border-bord-white-200 shadow-sm", children: /* @__PURE__ */ e("a", { href: "", className: "text-sm text-color-font-700 font-medium", children: "Cancel" }) }),
      /* @__PURE__ */ e(
        "button",
        {
          type: "submit",
          className: "flex px-[17px] py-[9px] justify-center items-center rounded-md bg-delhivery-color-100 shadow-sm",
          onClick: se,
          children: /* @__PURE__ */ e("span", { className: "text-sm text-color-font-100 font-medium", children: "Save" })
        }
      )
    ] }) })
  ] }) : null;
}, Re = () => {
  const t = de(), { getUserManager: s } = W(), r = ke(!1);
  return F(() => {
    if (r.current) return;
    (async () => {
      r.current = !0;
      const i = sessionStorage.getItem("issuer");
      if (!i) {
        console.error("Unable to find issuer.");
        const m = C.landing_page || "/";
        await t(m.startsWith("/") ? m : `/${m}`);
        return;
      }
      let c = s() || A.getUserManager();
      if (!c) {
        const m = new URL(window.location.href), u = m.host, h = m.protocol, k = C.clientId, N = `${h}//${u}/lib-auth-oauth-callback`, I = J.configOptions(i, k, N);
        c = new X(I), A.setUserManager(c);
      }
      try {
        await c.metadataService.getMetadata();
        const m = await c.signinRedirectCallback();
        m && m.access_token && (sessionStorage.setItem("access_token", m.access_token), O(c), localStorage.removeItem("loginInitiated"));
        const u = C.landing_page || "/";
        window.location.href = u.startsWith("/") ? u : `/${u}`;
      } catch (m) {
        console.error("OAuth callback error:", m), localStorage.removeItem("loginInitiated");
        const u = C.landing_page || "/";
        window.location.href = u.startsWith("/") ? u : `/${u}`;
      }
    })();
  }, []), null;
}, Ve = ({ children: t, fallback: s = /* @__PURE__ */ e("div", { children: "Loading..." }) }) => {
  const { isAuthenticated: r, isLoading: n } = W();
  return n ? /* @__PURE__ */ e(Q, { children: s }) : r ? /* @__PURE__ */ e(Q, { children: t }) : /* @__PURE__ */ e(Q, { children: s });
}, Ge = [
  /* @__PURE__ */ e(ne, { path: "/lib-auth-oauth-callback", element: /* @__PURE__ */ e(Re, {}) }, "oauth-callback"),
  /* @__PURE__ */ e(ne, { path: "/lib-user-profile", element: /* @__PURE__ */ e(Oe, {}) }, "user-profile")
];
class Ee {
  async expandRPT(s, r) {
    if (!sessionStorage.getItem("access_token"))
      throw new Error("401 Unauthorised");
    const n = new URLSearchParams();
    return n.set("grant_type", "urn:ietf:params:oauth:grant-type:uma-ticket"), n.set("ticket", s), (await $.post(
      r + "/protocol/openid-connect/token",
      n.toString(),
      {
        headers: {
          Authorization: "Bearer " + sessionStorage.getItem("access_token"),
          "Content-Type": "application/x-www-form-urlencoded"
        }
      }
    )).data;
  }
}
const Xe = new Ee();
export {
  Be as AuthProvider,
  Ge as AuthRoutes,
  Y as DEFAULT_HEADER_CONFIG,
  Pe as Header,
  Me as HeaderConfigProvider,
  Re as OAuthCallback,
  Ve as ProtectedRoute,
  Oe as UserProfile,
  C as configService,
  A as loginService,
  J as oidcConfigService,
  Xe as permissionService,
  ce as setupAuthInterceptor,
  W as useAuth,
  De as useHeaderConfig
};
